File: src/main/java/io/openfuture/openmessanger/repository/entity/GroupChat.java
package io.openfuture.openmessanger.repository.entity;

import java.time.LocalDateTime;
import java.time.ZonedDateTime;
import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor

@Data
@Entity
@Table(name = "group_chat")
public class GroupChat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "creator")
    private String creator;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "name", length = 255)
    private String name;

    @Column(name = "archived")
    private boolean archived = false;

    @Column(name = "archived_at")
    private LocalDateTime archivedAt;

    public GroupChat(final String creator, final LocalDateTime createdAt, final String name) {
        this.creator = creator;
        this.createdAt = createdAt;
        this.name = name;
    }

    public GroupChat(final Integer id) {
        this.id = id;
    }

}

File: src/main/kotlin/io/openfuture/openmessenger/repository/entity/GroupChat.kt
package io.openfuture.openmessenger.repository.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "group_chat")
class GroupChat() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null

    @Column(name = "creator")
    var creator: String? = null

    @Column(name = "created_at")
    var createdAt: LocalDateTime? = null

    @Column(name = "name", length = 255)
    var name: String? = null

    @Column(name = "archived")
    var archived = false

    @Column(name = "archived_at")
    var archivedAt: LocalDateTime? = null

    @OneToMany(mappedBy = "groupChat")
    var groupParticipants: List<GroupParticipant>? = null

    constructor(creator: String?, createdAt: LocalDateTime?, name: String?) : this() {
        this.creator = creator
        this.createdAt = createdAt
        this.name = name
    }

    constructor(id: Int?) : this() {
        this.id = id
    }
    
}